server/src/domain/usecases/createTask.ts

```diff
@@ -0,0 +1,15 @@
+import { Task, TaskProps } from '../entities/task'
+import { TaskRepository } from '../../infrastructure/persistance/taskRepository'
+
+export class CreateTask {
```

Comment:
the task repository here is storage layer. A storage layer is a layer in the infrastructure.

Reply:
Thanks, not sure if I did it well. I put use-cases in application layer